src/main/play-doc/general/MigrationGuide.md

```diff
@@ -33,18 +33,333 @@ In additional to disabling Elasticsearch via configuration, a service lookup mus
 
-```json
-    endpoints        = {
-      "angular-seed-play" = {
-        protocol  = "http"
-        bind-port = 0
-        services  = ["http:/myservice"]
+The [[request acl|RequestAclConfiguration]] in conjunction with [[dynamic proxy configuration|DynamicProxyConfiguration]] provides additional proxy mapping flexibility on top of what was previously provided by the service ports. This will allow operations to have a complete control on the proxy configuration.
+
+### Prerequisite
+
+Specify the ConductR target version in the `build.sbt`
+
+```
+BundleKeys.conductrTargetVersion := ConductrVersion.V1_2
+```
+
+The ConductR target version does not need to be declared in `bundle.conf` specified within bundle configuration override.
+
+### Migrating HTTP-based services
+
+Some HTTP based service definition can be migrated by supplying its [[request acl|RequestAclConfiguration]] equivalent, while others require a customised HAProxy configuration.
+
+#### Migrating services definition
+
+The following HTTP based service definition can be migrated by supplying its [[request acl|RequestAclConfiguration]] equivalent.
+
+##### Services mapped to root path
+
+Given the following example where the service URI was "http:/service-name", ConductR will interpret the first path component (`service-name`) as the service name for the purposes of service lookup. By default ConductR will then remove the first path component when rewriting the request. This means that your application or service will receive everything under the root of `/`.
+
+###### Migrating build.sbt
+
+Modify the `Endpoint` declared within the `BundleKeys.endpoints`.
+
+From:
+
+```
+BundleKeys.endpoints := Map(
+  "endpoint-label" -> Endpoint("http", 0, Set(URI("http:/service-name")))
+)
+```
+
+To:
+
+```
+BundleKeys.endpoints := Map(
+  "endpoint-label" -> Endpoint("http", 0, "service-name",
+    RequestAcl(Http("^/service-name".r -> "/"))
+  )
+)
+```
+
+###### Migrating bundle.conf
+
+Modify the `endpoints` section declared within the `bundle.conf` as such.
+
+From:
+
+```
+    endpoints         = {
+      "endpoint-label" = {
+        bind-protocol = "http"
+        bind-port     = 0
+        services      = ["http:/service-name]
+      }
+    }
+```
+
+To:
+
+```
+    endpoints         = {
+      "endpoint-label" = {
+        bind-protocol = "http"
+        bind-port     = 0
+        service-name  = "service-name"
+        acls          = [
+          {
+            http = {
+              requests = [
+                {
+                  path-beg = "/service-name"
+                  rewrite = "/"
+                }
+              ]
+            }
+          }
+        ]
+      }
+    }
+```
+
+
+
+##### Services mapped with preserved path
+
+
+Given the following service declaration, the applications or services expect the context root to be preserved when passing through ConductR's proxy, then the service URI can be instead expressed as "http:/service-name?preservePath"
```

Comment:
What about: "The `preservePath` query parameter on a service declaration indicated that the context root should be preserved when passing through ConductR's proxy. Given the service URI `http:/my-service?preservePath`, the application is accessible from the outside world on `http://{port-allocated-by-conductr}`.

This behaviour can now be declared with a request ACL."